woocommerce/woocommerce-gateway-stripe
# Revert getting the Stripe gateway from the list of enabled gateways

# Changes proposed in this Pull Request:

This PR reverts a change included in #1856, which changed `get_gateway()` to search for the Stripe gateway only in the list of enabled gateways.

This is necessary because we are creating an instance of `WC_Stripe_Intent_Controller` after the class declaration even when the gateway is not enabled... and in the `__construct` function we register this `add_action( 'wp', [ $this, 'maybe_process_upe_redirect' ] );` so when that action tries to get the gateway to check if its the UPE or not, it generates the Notice:
```
Notice: Undefined index: stripe in /var/www/html/wp-content/plugins/woocommerce-gateway-stripe/includes/class-wc-stripe-intent-controller.php on line 51
```

- Another solution would be to keep the change, check if the `stripe` gateway is present and if not return `null`.
- An even better solution would be not to instantiate this class if the Stripe gateway is not enabled.

# Testing instructions
- Checkout the branch: `develop`.
- Disable the Stripe Gateway, load any page and you should get the notice above in the logs.
- Checkout this branch.
- Load any page and you should NOT see the notice above.
